test/httpPipelineTests.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+import * as assert from "assert";
+import { HttpClient } from "../lib/httpClient";
+import { HttpPipeline } from "../lib/httpPipeline";
+import { HttpRequest } from "../lib/httpRequest";
+import { HttpResponse } from "../lib/httpResponse";
+import { userAgentRequestPolicyFactory } from "../lib/policies/userAgentRequestPolicyFactory";
```

Comment:
IMO let the impls of RequestPolicyFactory just be named e.g. userAgentPolicyFactory instead of userAgentRequestPolicyFactory. This is what we did in Java

Reply:
Okay.